contentcuration/contentcuration/utils/publish.py

```diff
@@ -319,20 +323,43 @@ def recurse_nodes(self, node, inherited_fields):  # noqa C901
             )
 
             if node.kind_id == content_kinds.EXERCISE:
-                exercise_data = process_assessment_metadata(node, kolibrinode)
+                exercise_data = process_assessment_metadata(node)
+                any_free_response = any(
+                    t == exercises.FREE_RESPONSE
+                    for t in exercise_data["assessment_mapping"].values()
+                )
+                generator_class = (
+                    QTIExerciseGenerator
+                    if any_free_response
+                    else PerseusExerciseGenerator
+                )
+
+                # If this exercise previously had a file generated by a different
+                # generator, make sure we clean it up here.
+                stale_presets = {
```

Comment:
Ensure we cleanup old files if an exercise now has free response or did previously have free response.